src/test/scala/uk/gov/nationalarchives/signcookies/LambdaTest.scala

```diff
@@ -0,0 +1,100 @@
+
+import java.time.Instant
+import java.util.UUID
+
+class LambdaTest extends AnyFlatSpec with BeforeAndAfterEach {
+
+  override def beforeEach(): Unit = {
+    wiremockKmsEndpoint.start()
+    wiremockKmsEndpoint.stubFor(post(urlEqualTo("/")))
+    keycloakMock.start()
+  }
+
+  override def afterEach(): Unit = {
+    wiremockKmsEndpoint.stop()
+    keycloakMock.stop()
+  }
+
+
+  "the lambda" should "return 401 if the token is expired" in {
+    val accessToken = createAccessToken(UUID.randomUUID().some, Instant.now().minusSeconds(3600))
+    val inputStream = getInputStream(accessToken)
+    val byteArrayCaptor: ArgumentCaptor[Array[Byte]] = ArgumentCaptor.forClass(classOf[Array[Byte]])
+
+    new Lambda().handleRequest(inputStream, outputStream(byteArrayCaptor), null)
+
+    decode[LambdaResponse](byteArrayCaptor.getValue.map(_.toChar).mkString).toOption.get.statusCode should equal(401)
+  }
+
+  "the lambda" should "return 401 if the token is not a valid jwt token" in {
+    val accessToken = "faketoken"
+    val inputStream = getInputStream(accessToken)
+    val byteArrayCaptor: ArgumentCaptor[Array[Byte]] = ArgumentCaptor.forClass(classOf[Array[Byte]])
+
+    new Lambda().handleRequest(inputStream, outputStream(byteArrayCaptor), null)
+
+    decode[LambdaResponse](byteArrayCaptor.getValue.map(_.toChar).mkString).toOption.get.statusCode should equal(401)
+  }
+
+  "the lambda" should "return 401 if the token has no user id" in {
+    val accessToken = createAccessToken(None)
+    val inputStream = getInputStream(accessToken)
+    val byteArrayCaptor: ArgumentCaptor[Array[Byte]] = ArgumentCaptor.forClass(classOf[Array[Byte]])
+
+    new Lambda().handleRequest(inputStream, outputStream(byteArrayCaptor), null)
+
+    decode[LambdaResponse](byteArrayCaptor.getValue.map(_.toChar).mkString).toOption.get.statusCode should equal(401)
+  }
+
+  "the lambda" should "return 200 if the token is valid" in {
+    val accessToken = createAccessToken(UUID.randomUUID().some)
+    val inputStream = getInputStream(accessToken)
+    val byteArrayCaptor: ArgumentCaptor[Array[Byte]] = ArgumentCaptor.forClass(classOf[Array[Byte]])
+
+    new Lambda().handleRequest(inputStream, outputStream(byteArrayCaptor), null)
+
+    decode[LambdaResponse](byteArrayCaptor.getValue.map(_.toChar).mkString).toOption.get.statusCode should equal(200)
+  }
+
+  "the lambda" should "return allowed origin localhost if the origin is localhost" in {
+    val accessToken = createAccessToken(UUID.randomUUID().some)
+    val inputStream = getInputStream(accessToken)
+    val byteArrayCaptor: ArgumentCaptor[Array[Byte]] = ArgumentCaptor.forClass(classOf[Array[Byte]])
+
+    new Lambda().handleRequest(inputStream, outputStream(byteArrayCaptor), null)
+
+    decode[LambdaResponse](byteArrayCaptor.getValue.map(_.toChar).mkString).toOption.get.statusCode should equal(200)
+  }
+
+  "the lambda" should "return allowed origin integration if the origin is integration" in {
```

Comment:
I didn't do them because I couldn't work out how to change the environment variables in a sane way but there probably should be some. I also thought that the Lambda class was getting a bit big so I've split off the `createResponse` and `createCookies` methods into their own class and merged them. 

We can't really test the `createCookies` method as it's mostly calling SDK stuff and the response is encrypted anyway.
I've added tests in for the different combinations of stage, origin and frontend url.

Reply:
The comment about testing `createCookies` was wrong, we can decode the policy so I've added in a test for the cookies policy.